### PR TITLE
ticdc: add binary_encoding_method to openapi v2

### PR DESCRIPTION
**Pull request by CharlesCheung96**
### First-time contributors' checklist 

- [x] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs-cn) that's required for repo owners to accept my contribution.

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions (in Chinese)](https://github.com/pingcap/docs-cn/blob/master/CONTRIBUTING.md#版本选择指南).

- [x] master (the latest development version)
- [ ] v7.4 (TiDB 7.4 versions)
- [ ] v7.3 (TiDB 7.3 versions)
- [ ] v7.2 (TiDB 7.2 versions)
- [x] v7.1 (TiDB 7.1 versions)
- [ ] v7.0 (TiDB 7.0 versions)
- [x] v6.5 (TiDB 6.5 versions)
- [ ] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

**Review by nongfushanquan — APPROVED** [review]

**Review by sdojjy — APPROVED** [review]

File: ticdc/ticdc-changefeed-config.md [hunk 1]
```diff
@@ -283,4 +289,19 @@ batching-max-messages=1000
...
+# 向下游存储服务保存数据变更记录的并发度，默认值为 16。
+worker-count = 16
+# 向下游存储服务保存数据变更记录的间隔，默认值为 2s。
+flush-interval = "2s"
+# 单个数据变更文件的字节数超过 `file-size` 时将其保存至存储服务中，默认值为 64MB。
+file-size = 67108864
+
+# 文件保留的时长，仅在 date-separator 配置为 day 时生效，默认值为 0，表示禁用文件清理。假设 `file-expiration-days = 1`，TiCDC 的清理行为是：在 2023/12/02 00:00:00 时刻，TiCDC 会清理 2023/12/01 之前（注意：不包括 2023/12/01）的文件。
+file-expiration-days = 0
+# 定时清理任务的运行周期，与 crontab 配置兼容，格式为 `Second | Minute | Hour | Dom | Month | DowOptional`，默认值为 "0 0 2 * * *"，表示每天晚上两点执行清理任务
```
Comment (qiancai) [inline ticdc/ticdc-changefeed-config.md:303]:
```suggestion
# 定时清理任务的运行周期，与 crontab 配置兼容，格式为 `Second | Minute | Hour | Dom | Month | DowOptional`，默认值为 "0 0 2 * * *"，表示每天凌晨两点执行清理任务
```

File: ticdc/ticdc-open-api-v2.md [hunk 2]
```diff
@@ -285,6 +285,8 @@ curl -X GET http://127.0.0.1:8300/api/v2/health
 | `level`          | `STRING` 类型，同步数据的一致性级别。（非必选）           |
 | `max_log_size`   | `UINT64` 类型，redo log 的最大值。（非必选）        |
 | `storage`        | `STRING` 类型，存储的目标地址。（非必选）              |
+| `use_file_backend`        | `BOOL` 类型，是否将 redo log 存储到本地文件中。（非必选）              |
+| `encoding_worker_num`        | `INT` 类型，redo 中编解码 worker 的数量。（非必选）              |`flush_worker_num`        | `INT` 类型，redo 中上传文件 worker 的数量。（非必选）              |
```
Comment (qiancai) [inline ticdc/ticdc-open-api-v2.md:289]:
```suggestion
| `encoding_worker_num`        | `INT` 类型，redo 模块中编解码 worker 的数量。（非必选）              
|`flush_worker_num`        | `INT` 类型，redo 模块中上传文件 worker 的数量。（非必选）              |
```

File: ticdc/ticdc-open-api-v2.md [hunk 3]
```diff
@@ -362,6 +365,17 @@ curl -X GET http://127.0.0.1:8300/api/v2/health
 | `partition` | `STRING` 类型，事件分发的目标 partition。    |
 | `topic`     | `STRING` 类型，事件分发的目标 topic。        |
 
+`sink.cloud_storage_config` 参数说明如下：
```
Comment (qiancai) [inline ticdc/ticdc-open-api-v2.md:368]:
```suggestion
`sink.cloud_storage_config` 参数说明如下：

```

File: ticdc/ticdc-server-config.md [hunk 4]
```diff
@@ -40,6 +40,8 @@ data-dir = ""
 gc-ttl = 86400 # 24 h
 tz = "System"
 cluster-id = "default"
+# 控制 GOGC Tuner 自动调节的最大内存阈值（计算单位为 byte）：设置较小的阈值会提高 GC 频率；设置较大的阈值会降低 GC 频率并使 TiCDC 进程占用更多的内存资源；超过阈值后 GOGC Tuner 会停止工作。默认值为 0，表示禁用 GOGC Tuner。
```
Comment (qiancai) [inline ticdc/ticdc-server-config.md:43]:
```suggestion
# 控制 GOGC Tuner 自动调节的最大内存阈值（单位为 byte）：设置较小的阈值会提高 GC 频率；设置较大的阈值会降低 GC 频率并使 TiCDC 进程占用更多的内存资源；超过阈值后 GOGC Tuner 会停止工作。默认值为 0，表示禁用 GOGC Tuner。
```

File: ticdc/ticdc-server-config.md [hunk 5]
```diff
@@ -40,6 +40,8 @@ data-dir = ""
...
+gc-tuner-memory-threshold = 0
```
Comment (qiancai) [inline ticdc/ticdc-server-config.md:44]: 多大算较小的阈值，可以提供一个例子吗

Comment (CharlesCheung96) [reply ticdc/ticdc-server-config.md:44]: 这里的参数是内存绝对值，所以跟最大可用内存有关系。 绝大部分情况下，这个参数都会保持为默认值。

File: ticdc/ticdc-changefeed-config.md [hunk 6]
```diff
@@ -283,4 +289,19 @@ batching-max-messages=1000
 batching-max-publish-delay=10
 # Pulsar Producer 发送消息的超时时间，默认 30 秒。
 send-timeout=30
+
+[sink.cloud-storage-config]
+# 向下游存储服务保存数据变更记录的并发度，默认值为 16。
+worker-count = 16
+# 向下游存储服务保存数据变更记录的间隔，默认值为 "2s"。
+flush-interval = "2s"
+# 单个数据变更文件的字节数超过 `file-size` 时将其保存至存储服务中，默认值为 67108864，即 64MB。
```
Comment (qiancai) [inline ticdc/ticdc-changefeed-config.md:298]:
```suggestion
# 单个数据变更文件的字节数超过 `file-size` 时将其保存至存储服务中，默认值为 67108864，即 64 MiB。
```

File: ticdc/ticdc-open-api-v2.md [hunk 7]
```diff
@@ -330,6 +333,7 @@ curl -X GET http://127.0.0.1:8300/api/v2/health
 | `terminator`            | `STRING` 类型，换行符，用来分隔两个数据变更事件。默认值为空，表示使用 `"\r\n"` 作为换行符。（非必选）                                                         |
 | `transaction_atomicity` | `STRING` 类型，事务一致性等级。（非必选）                                                                                            |
 | `only_output_updated_columns`             | `BOOLEAN` 类型，对于 MQ 类型的 Sink 中的 `canal-json` 和 `open-protocol`，表示是否只向下游同步有内容更新的列。默认值为 `false`。（非必选）                        |
+| `cloud_storage_config`             | storage sink 配置，用于设置同步数据到对象存储时自动清理历史数据的功能。（非必选） |
```
Comment (CharlesCheung96) [inline ticdc/ticdc-open-api-v2.md:336]:
```suggestion
| `cloud_storage_config`             | storage sink 配置。（非必选） |
```

File: ticdc/ticdc-open-api-v2.md [hunk 8]
```diff
@@ -285,6 +285,9 @@ curl -X GET http://127.0.0.1:8300/api/v2/health
 | `level`          | `STRING` 类型，同步数据的一致性级别。（非必选）           |
 | `max_log_size`   | `UINT64` 类型，redo log 的最大值。（非必选）        |
 | `storage`        | `STRING` 类型，存储的目标地址。（非必选）              |
+| `use_file_backend`        | `BOOL` 类型，是否将 redo log 存储到本地文件中。（非必选）              |
+| `encoding_worker_num`        | `INT` 类型，redo 模块中编解码 worker 的数量。（非必选）
```
Comment (ran-huang) [inline ticdc/ticdc-open-api-v2.md:289]:
```suggestion
| `encoding_worker_num`        | `INT` 类型，redo 模块中编解码 worker 的数量。（非必选）|
```

File: ticdc/ticdc-open-api-v2.md [hunk 9]
```diff
@@ -340,12 +344,13 @@ curl -X GET http://127.0.0.1:8300/api/v2/health
...
-| `include_commit_ts` | `BOOLEAN` 类型，是否在 CSV 行中包含 commit-ts。默认值为 `false`。 |
-| `null`              | `STRING` 类型，如果这一列是 null，那这一列该如何表示。默认是用 `\N` 来表示。 |
-| `quote`             | `STRING` 类型，用于包裹字段的引号字符。空值代表不使用引号字符。默认值为 `"`。   |
+| 参数名                    | 说明                                              |
+|:-------------------------|:------------------------------------------------|
+| `delimiter`              | `STRING` 类型，字段之间的分隔符。必须为 ASCII 字符，默认值为 `,`。     |
+| `include_commit_ts`      | `BOOLEAN` 类型，是否在 CSV 行中包含 commit-ts。默认值为 `false`。 |
+| `null`                   | `STRING` 类型，如果这一列是 null，那这一列该如何表示。默认是用 `\N` 来表示。 |
+| `quote`                  | `STRING` 类型，用于包裹字段的引号字符。空值代表不使用引号字符。默认值为 `"`。   |
+| `binary_encoding_method` | `STRING` 类型，二进制类型数据的编码方式，可选 'base64' 或 'hex'。默认值为 'base64'。   |
```
Comment (ran-huang) [inline ticdc/ticdc-open-api-v2.md:353]:
```suggestion
| `binary_encoding_method` | `STRING` 类型，二进制类型数据的编码方式，可选 `"base64"` 或 `"hex"`。默认值为 `"base64"`。   |
```

File: ticdc/ticdc-open-api-v2.md [hunk 10]
```diff
@@ -362,6 +367,18 @@ curl -X GET http://127.0.0.1:8300/api/v2/health
...
 
+`sink.cloud_storage_config` 参数说明如下：
+
+| 参数名         | 说明                                |
+|:------------|:----------------------------------|
+| `worker_count`   | `INT` 类型，向下游存储服务保存数据变更记录的并发度。 |
+| `flush_interval`   | `STRING` 类型，向下游存储服务保存数据变更记录的间隔。 |
+| `file_size`   | `INT` 类型，单个数据变更文件的字节数超过 `file-size` 时将其保存至存储服务中。 |
+| `file_expiration_days`   | `INT` 类型，文件保留的时长。|
+| `file_cleanup_cron_spec`   | `STRING` 类型，定时清理任务的运行周期，与 crontab 配置兼容，格式为 `Second | Minute | Hour | Dom | Month | DowOptional`。|
```
Comment (ran-huang) [inline ticdc/ticdc-open-api-v2.md:378]: 这里变成了图上这样，转义试试能不能正常显示
image

```suggestion
| `file_cleanup_cron_spec`   | `STRING` 类型，定时清理任务的运行周期，与 crontab 配置兼容，格式为 `Second \| Minute \| Hour \| Dom \| Month \| DowOptional`。|
```

File: ticdc/ticdc-changefeed-config.md [hunk 11]
```diff
@@ -283,4 +289,19 @@ batching-max-messages=1000
...
+# 向下游存储服务保存数据变更记录的并发度，默认值为 16。
+worker-count = 16
+# 向下游存储服务保存数据变更记录的间隔，默认值为 "2s"。
+flush-interval = "2s"
+# 单个数据变更文件的字节数超过 `file-size` 时将其保存至存储服务中，默认值为 67108864，即 64 MiB。
+file-size = 67108864
+
+# 文件保留的时长，仅在 date-separator 配置为 day 时生效，默认值为 0，表示禁用文件清理。假设 `file-expiration-days = 1` 且 `file-cleanup-cron-spec = "0 0 0 * * *"`，TiCDC 将在每天 00:00:00 时刻清理已保存超过 24 小时的文件。例如，2023/12/02 00:00:00 将清理 2023/12/01 之前（注意：不包括 2023/12/01）的文件。
+file-expiration-days = 0
+# 定时清理任务的运行周期，与 crontab 配置兼容，格式为 `Second | Minute | Hour | Dom | Month | DowOptional`，默认值为 "0 0 2 * * *"，表示每天凌晨两点执行清理任务
```
Comment (qiancai) [inline ticdc/ticdc-changefeed-config.md:303]:
```suggestion
# 定时清理任务的运行周期，与 crontab 配置兼容，格式为 `<Second> <Minute> <Hour> <Dom> <Month> <DowOptional>`，默认值为 "0 0 2 * * *"，表示每天凌晨两点执行清理任务
```

File: ticdc/ticdc-open-api-v2.md (hunk 10 above)
Comment (qiancai) [reply ticdc/ticdc-open-api-v2.md:378]:
```suggestion
| `file_cleanup_cron_spec`   | `STRING` 类型，定时清理任务的运行周期，与 crontab 配置兼容，格式为 `<Second> <Minute> <Hour> <Dom> <Month> <DowOptional>`。|
```

File: ticdc/ticdc-open-api-v2.md [hunk 12]
```diff
@@ -362,6 +367,18 @@ curl -X GET http://127.0.0.1:8300/api/v2/health
...
+
+| 参数名         | 说明                                |
+|:------------|:----------------------------------|
+| `worker_count`   | `INT` 类型，向下游存储服务保存数据变更记录的并发度。 |
+| `flush_interval`   | `STRING` 类型，向下游存储服务保存数据变更记录的间隔。 |
+| `file_size`   | `INT` 类型，单个数据变更文件的字节数超过 `file-size` 时将其保存至存储服务中。 |
+| `file_expiration_days`   | `INT` 类型，文件保留的时长。|
+| `file_cleanup_cron_spec`   | `STRING` 类型，定时清理任务的运行周期，与 crontab 配置兼容，格式为 `<Second> <Minute> <Hour> <Day of the month> <Month> <Day of the week (Optional)>`。|
+| `flush_concurrency`   | `INT` 类型，上传单个文件的并发数。|
+
```
Comment (ran-huang) [inline ticdc/ticdc-open-api-v2.md:380]:
```suggestion
```

**Review by ran-huang — APPROVED** [review]

File: ticdc/ticdc-changefeed-config.md [hunk 13]
```diff
@@ -283,4 +289,19 @@ batching-max-messages=1000
...
 send-timeout=30
+
+[sink.cloud-storage-config]
+# 向下游存储服务保存数据变更记录的并发度，默认值为 16。
+worker-count = 16
+# 向下游存储服务保存数据变更记录的间隔，默认值为 "2s"。
+flush-interval = "2s"
+# 单个数据变更文件的字节数超过 `file-size` 时将其保存至存储服务中，默认值为 67108864，即 64 MiB。
+file-size = 67108864
+
```
Comment (qiancai) [inline ticdc/ticdc-changefeed-config.md:300]:
```suggestion
```

**Pull request merged**